src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -1,652 +1,399 @@
       if (path && !nameControl.value) {
         const name = path.split('/').pop();
         nameControl.setValue(name);
       }
 
       if (!this.stripACLWarningSent) {
         this.ws.call('filesystem.acl_is_trivial', [path]).pipe(untilDestroyed(this)).subscribe((res) => {
-          if (!res && !entityForm.formGroup.controls['acl'].value) {
+          if (!res && !this.form.get('acl').value) {
             this.stripACLWarningSent = true;
             this.showStripAclWarning();
           }
         });
       }
     });
 
-    const pathControl = entityForm.formGroup.controls['path'];
-    entityForm.formGroup.controls['acl'].valueChanges.pipe(debounceTime(100)).pipe(untilDestroyed(this)).subscribe((res) => {
-      if (!res && pathControl.value && !this.stripACLWarningSent) {
-        this.ws.call('filesystem.acl_is_trivial', [pathControl.value]).pipe(untilDestroyed(this)).subscribe((res) => {
-          if (!res) {
-            this.stripACLWarningSent = true;
-            this.showStripAclWarning();
-          }
-        });
+    this.form.get('acl').valueChanges.pipe(debounceTime(100)).pipe(untilDestroyed(this)).subscribe((res) => {
+      if (!res && pathFormControl.value && !this.stripACLWarningSent) {
+        this.ws.call('filesystem.acl_is_trivial', [pathFormControl.value])
+          .pipe(untilDestroyed(this)).subscribe((res) => {
+            if (!res) {
+              this.stripACLWarningSent = true;
+              this.showStripAclWarning();
+            }
+          });
       }
     });
+  }
 
-    setTimeout(() => {
-      if (entityForm.formGroup.controls['timemachine'].value) {
-        this.isTimeMachineOn = true;
-      }
-    }, 700);
+  showStripAclWarning(): void {
+    this.dialog.confirm({
+      title: helptextSharingSmb.stripACLDialog.title,
+      message: helptextSharingSmb.stripACLDialog.message,
+      hideCheckBox: true,
+      buttonMsg: helptextSharingSmb.stripACLDialog.button,
+      hideCancel: true,
+    }).pipe(untilDestroyed(this)).subscribe();
+  }
 
-    entityForm.formGroup.controls['purpose'].valueChanges.pipe(untilDestroyed(this)).subscribe((res) => {
-      this.clearPresets();
-      for (const item in this.presets[res].params) {
-        this.presetFields.push(item as keyof SmbShare);
-        const ctrl = entityForm.formGroup.controls[item];
-        if (ctrl && item !== 'auxsmbconf') {
-          ctrl.setValue(this.presets[res].params[item as keyof SmbShare]);
-          ctrl.disable();
-        }
+  clearPresets(): void {
+    for (const item of this.presetFields) {
+      this.form.get(item).enable();
+    }
+    this.presetFields = [];
+  }
+
+  setSmbShareForEdit(smbShare: SmbShare): void {
+    this.existingSmbShare = smbShare;
+    this.hostsAllowOnLoad = smbShare.hostsallow ? [...smbShare.hostsallow] : [];
+    this.mangle = smbShare.aapl_name_mangling;
+    this.hostsDenyOnLoad = smbShare.hostsdeny ? [...smbShare.hostsdeny] : [];
+    this.isTimeMachineOn = smbShare.timemachine;
```

Comment:
You save a bunch of fields for values before edit, but you also have `existingSmbShare` that you can use.